pkg/haproxy/handler.go

```diff
@@ -34,14 +35,17 @@ type Handler struct {
 
 // Init initializes a new Handler
 func Init(updates chan state.ClusterState, config config.Config) (*Handler, error) {
-	parsedTemplate, err := template.ParseFiles(config.HAProxyTemplatePath)
+	rawTemplateString, err := ioutil.ReadFile(config.HAProxyTemplatePath)
 	if err != nil {
 		return nil, err
 	}
-	parsedTemplate.Funcs(template.FuncMap{
+	parsedTemplate, err := template.New("template").Funcs(template.FuncMap{
 		"StringJoin": strings.Join,
-		"Replace": func (s, old, new string) string { return strings.Replace(s, old, new, -1) },
-	})
+		"replace":    func(s, old, new string) string { return strings.Replace(s, old, new, -1) },
```

Comment:
The name of the template function should be `Replace`, since `StringJoin` is also capitalised and Go likes capitalised words for exported functions.

Reply:
Functions used in templates are not exported, they are put in scope of the global function map. Since the other functions in the global function map are also non-capitalized I felt it was good practice to conform to that convention.